src/Scaler/Services/CosmosDbFactory.cs

```diff
@@ -9,14 +10,19 @@ internal sealed class CosmosDbFactory
         // maintain a single instance of CosmosClient per lifetime of the application.
         private readonly ConcurrentDictionary<string, CosmosClient> _cosmosClientCache = new();
 
-        public CosmosClient GetCosmosClient(string connection)
+        public CosmosClient GetCosmosClient(string endpoint)
         {
-            return _cosmosClientCache.GetOrAdd(connection, CreateCosmosClient);
+            return _cosmosClientCache.GetOrAdd(endpoint, CreateCosmosClient);
         }
 
-        private CosmosClient CreateCosmosClient(string connection)
+        //private CosmosClient CreateCosmosClient(string connection)
+        private CosmosClient CreateCosmosClient(string endpoint)
```

Comment:
We are removing support for connection-string based cosmos-client. Can we support both connection strings and pod identity depending on the trigger metadata?

Reply:
Fixed